### PR TITLE
Use vector instead of set to keep the order of the opt passes

### PR DESCRIPTION
**Pull request by houseroad**
Use set is dangerous. The order is important for some passes. Use vector to replace set.

### CODE REVIEW

File: onnx/optimizer/pass_manager.h [hunk 1]
```diff
@@ -35,7 +35,7 @@ class GeneralPassManager : public PassManager {
   std::shared_ptr<PassManagerAnalysis> run(Graph& graph) override;
 
  protected:
-  std::set<std::shared_ptr<Pass>> passes;
+  std::vector<std::shared_ptr<Pass>> passes;
```
Comment (bddppq) [inline onnx/optimizer/pass_manager.h]: Maybe add a comment saying the order we add the passes matters, so people won't accidentally change it back to set.

Comment (houseroad) [reply onnx/optimizer/pass_manager.h]: Sure, will add that

**Comment by ArmenAg** [comment]
My bad. Yes it should be vector. But if we really want to insure the constraint that split is called last, we should implement it the pass manager.

**Comment by houseroad** [comment]
@ArmenAg yeah, we can add some mechanism to enforce some invariance.

**Comment by ArmenAg** [comment]
@houseroad Agreed. This can also break the fixed point optimization, since it might call split multiple times. We could make a new type of pass. Maybe call it something like SingularPass that is guaranteed to only run once throughout the lifetime of the optimization. It would also include a field that constrains where in the optimization pass has to run (beginning, end, any).

**Comment by houseroad** [comment]
@ArmenAg You can add this support in another diff. :-)

**Pull request merged**